ui/apps/platform/src/Containers/NetworkGraph/components/CIDRFormModal.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useEffect, useState } from 'react';
+import { FormikProvider, useFormik } from 'formik';
+import { Alert, Bullseye, Spinner, Modal, Button, Flex } from '@patternfly/react-core';
+import * as Yup from 'yup';
+
+import { isValidCidrBlock } from 'utils/urlUtils';
+import {
+    fetchCIDRBlocks,
+    deleteCIDRBlock,
+    postCIDRBlock,
+    patchCIDRBlock,
+} from 'services/NetworkService';
+import { getHasDuplicateCIDRNames, getHasDuplicateCIDRAddresses } from './cidrFormUtils';
+import DefaultCIDRToggle from './DefaultCIDRToggle';
+import CIDRForm, { emptyCIDRBlockRow, CIDRBlockEntity, CIDRBlockEntities } from './CIDRForm';
+
+const validationSchema = Yup.object().shape({
+    entities: Yup.array().of(
+        Yup.object().shape({
+            entity: Yup.object().shape({
+                name: Yup.string().trim().required('CIDR name is required.'),
+                cidr: Yup.string()
+                    .trim()
+                    .test('valid-cidr-format', 'CIDR address format is invalid.', (value) => {
+                        return isValidCidrBlock(value);
+                    })
+                    .required('CIDR address is required.'),
+            }),
+        })
+    ),
+});
+
+type FormCalloutType = 'success' | 'danger' | 'none';
+
+const emptyFormCallout = {
+    type: 'none' as FormCalloutType,
+    message: '',
+};
+
+type CIDRFormModalProps = {
+    selectedClusterId: string;
+    isOpen: boolean;
+    onClose: () => void;
+    // updateNetworkNodes: () => void;
```

Comment:
this will go in once the node updates section goes in since it also deals w refetching